### PR TITLE
  MMDP-521 Manage National Level SVG Matrix

### PR DESCRIPTION
**Pull request by jomasim**
**What does this PR do?**
- Allow an administrator to manage national level svg in content management system

Summarise the main tasks handled in the pull request

- [x] Upload a national level svg file
- [x] List country maps
- [x] List State maps
- [x] Edit State Id 

**How should this be manually tested?**

- Fetch and checkout to ` ft-manage-national-svg-521`
- Run the server `yarn run dev-server`

1. **Adding a new country map  svg file**
    - On the browser, enter `http://localhost:<port>/matrix/country`

Screenshot 2019-05-08 at 11 14 38

2.  **Listing states for a specific country**
     - Click on the **View**  button to list states for a specific country

Screenshot 2019-05-08 at 11 21 51

3. **Updating state ID**
      - Click on **Edit** button to edit state ID 

Screenshot 2019-05-08 at 11 23 05

4. **Updating country svg map** 
     - Upload a new country svg map from the country states listing page

Screenshot 2019-05-08 at 11 26 20

**NB**  Updating the country map svg is meant to **override the existing states** associated with the country.

**Any background context you want to provide?**

None

**What are the relevant Jira issues?**

[MMDP-521](https://viisaus.atlassian.net/secure/RapidBoard.jspa?rapidView=3&projectKey=MMDP&modal=detail&selectedIssue=MMDP-521&quickFilter=5)

**Screenshots (if appropriate)**
Check the screenshots shared above

**Questions:**

If something is unclear or you want some questions to be addressed by your peers, mention them here

### CODE REVIEW

File: client/__tests__/containers/Matrix/state.test.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { StateMap } from '../../../containers/Matrix/StateMap';
+
+let wrapper;
+let instance;
+let stateProps;
+
+describe('test StateBoundaries container container', () => {
```
Comment (Nicanor008) [inline client/__tests__/containers/Matrix/state.test.js]: Some Repetition here *container container*

File: client/__tests__/containers/Matrix/state.test.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        uniqueId: 'STL4343',
+        name: 'Lagos',
+        path: 'M434343',
+      },
+    ];
+    expect(instance.getStateMaps(data)).toEqual([
+      { id: 'STL4343', name: 'Lagos', path: 'M434343' },
+    ]);
+  });
+  // it('should update refresh to false on ComponentDidUpdate', () => {
```
Comment (FahdJamy) [inline client/__tests__/containers/Matrix/state.test.js]: Hey Joma. Great work
But I just think you might want to just delete out these commented out lines of code.

File: client/__tests__/containers/Matrix/state.test.js (hunk 1 above)
Comment (jomasim) [reply client/__tests__/containers/Matrix/state.test.js]: @Nicanor008 I have resolved this

File: package.json [hunk 3]
```diff
@@ -34,16 +34,23 @@
     "babel-preset-react-app": "^7.0.0",
     "bcrypt": "^3.0.5",
```
Comment (shemogumbe) [inline package.json]: Rebase with develop to remove bcrypt and only use bcryptjs, this package was removed long ago for causing compatibility problems

Comment (jomasim) [reply package.json]: @shemogumbe it is now fixed

**Pull request merged**